files/common/var/lib/delphix-platform/clear-sharenfs

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/python3
+
+import subprocess
+
+lines = subprocess.check_output(
+    "zfs list -t filesystem -r -H -o name,sharenfs,mounted domain0",
+    shell=True).decode('utf-8').splitlines()
+datasets = [line.split('\t') for line in lines]
+# Filter out datasets with sharenfs == "off" and sharenfs == "-"
+datasets = [ds for ds in datasets if ds[1] not in ('off', '-')]
+
+# We only clear sharenfs for datasets that are mounted as in some cases
+# zfs can mount datasets while clearing their sharenfs property and mount
+# children without mounting the parent datasets first. Given that ZFS
+# automatically creates directories for mountpoints, mounting children
+# before the parents would end up creating sub-directories under their
+# parent's mountpoints and so parents would fail to mount later. This
+# is something we really want to avoid, so as a precaution we make sure
+# that all the datasets that we manipulate are already mounted.
+unmounted_datasets = [ds for ds in datasets if ds[2] != 'yes']
+datasets = [ds for ds in datasets if ds not in unmounted_datasets]
+
+if unmounted_datasets:
+    print("WARNING: Not clearing sharenfs property for the following %d "
+          "datasets because they are unmounted:" % len(unmounted_datasets))
+    for ds in unmounted_datasets:
+        print("  %s" % ds[0])
+
+if datasets:
+    #
+    # We must use zfs inherit to clear the sharenfs property instead
+    # of setting sharenfs=off as doing the latter would set the SOURCE
+    # of the sharenfs property to "local" instead of "default". The
+    # app-stack shares timeflows by setting sharenfs on the timeflows and
+    # letting the setting propagate to its children, however this
+    # would not longer work if SOURCE for the sharenfs property is set to
+    # "local" for the children.
```

Comment:
What you have works, and is a huge improvement. Just thinking out loud: If the children's `sharenfs` properties are already set to `inherit`, then theoretically, we don't need to set those again to `inherit`, and presumably, there are many more of those than there are parents. I wonder if a more efficient approach could be to simply set `sharenfs` to `off` on those datasets that already have a `SOURCE` of `local`, and ignore any whose source is `inherit` given that they'll automatically get that setting.

Reply:
Yeah, this is similar to George's suggestion. I think filtering only datasets that have `SOURCE` set to `local` is reasonable and should give us the benefit of less calls to spa_sync. I'll try that out.